### PR TITLE
fix: build errors in sanitizers daily workflow

### PR DESCRIPTION
**Pull request by kostasrim**
Somehow, the sanitizers build triggered `-werror=deprecated` https://github.com/dragonflydb/dragonfly/actions/runs/10505992538/job/29104705713#step:7:941 for  `error: implicit capture of 'this' via '[=]'`. This is solved by disabling the warning for that block (see comments)

* disable warning for the problematic block in list_family

### CODE REVIEW

**Comment by kostasrim** [comment]
Sanitizers build https://github.com/dragonflydb/dragonfly/actions/runs/10511902823

File: src/server/list_family_test.cc [hunk 1]
```diff
@@ -796,7 +796,7 @@ TEST_F(ListFamilyTest, BLMoveRings) {
   for (int i = 0; i < 10; i++) {
     auto key1 = to_string(i);
     auto key2 = to_string(i + 1);
-    fibers.emplace_back(pp_->at(i % 3)->LaunchFiber([=]() {
```
Comment (kostasrim) [inline src/server/list_family_test.cc:799]: FFS, it launces a fiber

File: src/server/list_family_test.cc [hunk 2]
```diff
@@ -793,13 +793,20 @@ TEST_F(ListFamilyTest, BLMoveSimultaneously) {
 
 TEST_F(ListFamilyTest, BLMoveRings) {
   vector<fb2::Fiber> fibers;
+  fibers.reserve(10);
+#pragma GCC diagnostic push
+// We compile this code both with C++17 and C++20 and if you capture
+// by [=, this] it becomes an error on C++17 and if you capture
+// by [=] it becomes and error in C++20
+#pragma GCC diagnostic ignored "-Wdeprecated"
```
Comment (dranikpg) [inline src/server/list_family_test.cc:801]: we should post this comment to some r/cpprant or rust subreddit 😆

**Review by dranikpg — APPROVED** [review]

File: src/server/list_family_test.cc (hunk 2 above)
Comment (kostasrim) [reply src/server/list_family_test.cc:801]: Dude you have no idea how annoyed I was

**Pull request merged**